src/main/java/Grapie/Functions/Parser.java

```diff
@@ -0,0 +1,89 @@
+package Grapie.Functions;
+
+
+import Grapie.Commands.CommandTypes;
+import Grapie.Exceptions.ErrorMsg;
+import Grapie.Exceptions.GrapieExceptions;
+
+public class Parser {
+    public Parser() {
+        //deals with making sense of the user command
+    }
+
+    /**
+     * Check if a valid number is inputted in String form.
+     *
+     * @param numStr the string to be checked.
+     * @return Boolean stating if the String is a valid number.
+     */
+    public static boolean isNumber(String numStr) {
+        try {
+            Integer.parseInt(numStr);
+            return true;
+        } catch (NumberFormatException e) {
+            return false;
+        }
+    }
+
+    /**
+     * Returns the corresponding command for the user's input.
+     * Aims to sort user command into List, Delete, Done and Add.
+     * Add consist of everything else, including illegal commands, which will be handled in the Grapie
+     * .TaskList class.
+     *
+     * @param command The user's input.
+     * @return Returns a string stating if its a list, done, delete, or add (todo, event, deadline) command.
+     */
+    public CommandTypes.Commands parseCommand(String command) throws GrapieExceptions {
```

Comment:
Can you think of ways to improve SLAP here? (at this moment, it is OK not to change it, but be aware so that you can SLAP better in your TP code)